### PR TITLE
disable telemetry in tests

### LINKED ISSUE #292: Our tests should not send any telemetry.
I think our tests are sending telemetry.

### PR DESCRIPTION
**Pull request by PSeitz**
closes #292

### Context / purpose
Describe the context and purpose of this PR. Link related issue(s) if any.

### Description
Describe the proposed changes made in this PR.

### How was this PR tested?
Describe how you tested this PR.

### Checklist
*Remove or complete this list if required.*
- [ ] tested locally
- [ ] added unit tests
- [ ] included documentation

### CODE REVIEW

**Comment by PSeitz** [comment]
@fulmicoton I couldn't find a call from the tests to telemetry

File: quickwit-telemetry/src/sender.rs [hunk 1]
```diff
@@ -280,7 +280,7 @@ impl TelemetrySender {
 
 /// Check to see if telemetry is enabled.
 pub fn is_telemetry_enabled() -> bool {
-    std::env::var_os(crate::DISABLE_TELEMETRY_ENV_KEY).is_none()
+    !cfg!(test) || std::env::var_os(crate::DISABLE_TELEMETRY_ENV_KEY).is_none()
```
Comment (evanxg852000) [inline quickwit-telemetry/src/sender.rs:283]: I think the construct `cfg(test)` is not valid during integration test.
Can you please add 
```rust
    cmd.env(
        quickwit_telemetry::DISABLE_TELEMETRY_ENV_KEY,
        "disable-for-tests",
    )
```
to `tests::helpers::make_command`

Comment (PSeitz) [reply quickwit-telemetry/src/sender.rs:283]: we could also use debug_assertions
https://doc.rust-lang.org/reference/conditional-compilation.html#debug_assertions

Comment (fulmicoton) [reply quickwit-telemetry/src/sender.rs]: Ooops I think I already addressed this ticket by setting the env key in the test.

**Comment by fulmicoton** [comment]
https://github.com/quickwit-inc/quickwit/blob/main/quickwit-cli/tests/helpers.rs#L82-L86

**Comment by evanxg852000** [comment]
I was suggesting adding it to make_command as well https://github.com/quickwit-inc/quickwit/blob/main/quickwit-cli/tests/helpers.rs#L73

**Pull request closed without merging**